cfpb/capital-framework
# Updating cf-pagination to use the atomic structure from cfgov-refresh

This is the first update using the atomic code organization. It serves as an example of how each component should be converted.

## Additions

- Pagination molecule stylesheet

## Changes

- Updated documentation to show updated code examples
- Usage file for gh-pages

## Testing

- Install CF Sandbox https://github.com/jimmynotjim/capital-framework-sandbox#installation
- Click around to ensure the local server is rendering the sandbox correctly
- Stop the server (ctrl+c)
- In your local CF repo, checkout this branch and run `npm run cf-link`
- In your local CF Sandbox checkout the `cf-pagination-example` branch
- In your local CF Sandbox run `npm run cf-link && npm run build && npm start`
- Navigate to `http://localhost:3000/components/cf-pagination/`
- The _Original Documentation Markup_ should look mostly unstyled, the _Atomic Documentation Markup_ should look like the screenshot below, the _Live CFGov Markup_ should look completely broken (this is expected at the moment).
- To revert the cf-pagination changes in your local CF Sandbox run `npm unlink cf-pagination && npm install cf-pagination`

## Review

- @cfpb/front-end-team-admin 

## Screenshots

![screen shot 2016-06-21 at 4 48 57 pm](https://cloud.githubusercontent.com/assets/1280430/16247593/11d48e14-37d0-11e6-92b7-42fc4a6d202c.png)

## Notes

- It's difficult to test live code at the moment because these styles already exist in cfgov-refresh and testing in another project requires updating the project's markup. ~~A sandbox to test in is coming to make things easier.~~ It's happening!!!! https://jimmynotjim.github.io/capital-framework-sandbox/components/cf-pagination/
- `npm link` can be weird some times. I've found even when I have the new linked module in my `node_modules` dir I will still get the old styles when running `gulp build`. Making sure the local server isn't running and running the build task a few times seems, then restarting the server seems to fix it. ¯\_(ツ)_/¯ 
- The Sandbox currently includes base examples of cf-vars, cf-expandables, cf-pagination, and cf-buttons. All the other components need to have base examples included, and then each one can be updated with atomic examples as they are picked up.
- To find the atomic versions in cfgov-refresh, navigate to `cfgov/unprocessed/css` and check the `/atoms`, `/molecules`, and `/organisms` folders. Also be sure to check `cf-enhancements` for one-off updates that may or may not need to be included in the conversion.

## Todos

__ATOMIZE ALL THE THINGS!__

![shrunken-kids](https://cloud.githubusercontent.com/assets/1280430/16247510/ad0620ec-37cf-11e6-97ed-d43fa89f9d14.jpg)

## Checklist
